applications/dashboard/modules/class.sitetotalsmodule.php

```diff
@@ -2,58 +2,144 @@
 /**
  * Site totals module.
  *
- * @copyright 2009-2019 Vanilla Forums Inc.
+ * @copyright 2009-2020 Vanilla Forums Inc.
  * @license GPL-2.0-only
  * @package Dashboard
  * @since 2.0
  */
 
+use Vanilla\Scheduler\SchedulerInterface;
+
 /**
- * Site-wide stats.
+ * Class SiteTotalsModule
  */
 class SiteTotalsModule extends Gdn_Module {
 
+    /** @var int */
+    const CACHE_TTL = 43200;
+
+    /** @var int */
+    const RECALCULATE_INTERVAL = 900;
+
+    /** @var int */
+    const LOCK_INTERVAL = 60;
+
+    /** @var string */
+    const CACHE_KEY = 'module.sitetotals';
+
+    /** @var string */
+    const COUNTS_KEY = self::CACHE_KEY.'.counts';
+
+    /** @var string */
+    const LOCK_KEY = self::CACHE_KEY.'.lock';
+
+    /** @var string */
+    const RECALCULATE_KEY = self::CACHE_KEY.'.recalculate';
+
+    /**
+     * SiteTotalsModule constructor.
+     */
     public function __construct() {
         parent::__construct();
         $this->_ApplicationFolder = 'dashboard';
     }
 
+    /**
+     * @return string
+     */
     public function assetTarget() {
         return 'Panel';
     }
 
-    protected function _GetData() {
-        $px = Gdn::database()->DatabasePrefix;
-        $sql = "show table status where Name in ('{$px}User', '{$px}Discussion', '{$px}Comment')";
+    /**
+     * Get total counts
+     *
+     * @return mixed
+     */
+    public function getTotals() {
+        // check recalculate flag
+        $recalculateFlag = Gdn::cache()->get(self::RECALCULATE_KEY);
 
-        $result = ['User' => 0, 'Discussion' => 0, 'Comment' => 0];
-        foreach ($result as $name => $value) {
-            $result[$name] = $this->getCount($name);
+        if ($recalculateFlag === Gdn_Cache::CACHEOP_FAILURE) {  // expired
+            $this->tryRecalculate();
         }
-        $this->setData('Totals', $result);
+
+        // get totals from cache
+        $totals = Gdn::cache()->get(self::COUNTS_KEY);
+
+        return $totals;
     }
 
-    protected function getCount($table) {
-        // Try and get the count from the cache.
-        $key = "$table.CountRows";
-        $count = Gdn::cache()->get($key);
-        if ($count !== Gdn_Cache::CACHEOP_FAILURE) {
-            return $count;
+    /**
+     * Attempt to implement lock system. A failure likely means the
+     * cache key already exists, which would mean the lock is already in place.
+     */
+    private function tryRecalculate() {
```

Comment:
This method is returning a boolean on some case, nothing in others. If you refactor it slightly to make the "happy path" lead to one place without branching and `return true` in there, you can put a trailing `return false` at the end and make it a `: boolean` signature method.

So do:

```
$lock = Gdn::cache()->get(self::LOCK_KEY);

if ($lock !== Gdn_Cache::CACHEOP_SUCCESS) { 
    $added = Gdn::cache()->add(self::LOCK_KEY, mt_rand(0, 999999), [Gdn_Cache::FEATURE_EXPIRY => self::LOCK_INTERVAL]);

    if ($added) {
        /** @var Vanilla\Scheduler\SchedulerInterface $scheduler */
        $scheduler = Gdn::getContainer()->get(Vanilla\Scheduler\SchedulerInterface::class);
        $scheduler->addJob(
            Vanilla\Scheduler\Job\CallbackJob::class,
            [
                "callback" => function () {
                    $this->getAllCounts();
                }
            ]
        );
        return true;
    }
}

return false;
```